src/expr.cpp

```diff
@@ -1642,9 +1642,9 @@ llvm::Value *lEmitLogicalOp(BinaryExpr::Op op, Expr *arg0, Expr *arg1, FunctionE
     // evaluate both sides and then the logical operator in that case.
     // FIXME: not sure what we should do about vector types here...
     bool shortCircuit =
-        (EstimateCost(arg1) > PREDICATE_SAFE_IF_STATEMENT_COST || SafeToRunWithMaskAllOff(arg1) == false ||
-         CastType<VectorType>(type0) != NULL || CastType<VectorType>(type1) != NULL);
-    if (shortCircuit == false) {
```

Comment:
Generating always if conditions for vectors in not a good idea - I think safety check should work the same way it works for scalars. I.e. if no unsafe operations involved (pointer access, fp trapping operations), we should do the same as for scalars.

Reply:
Ok, I see that we check mask and etc for varying and need to do the same for vectors, so current fix is good enough.